docs/artifacts.md

```diff
@@ -0,0 +1,97 @@
+Artifact Versioning and Tagging (1.0.0)
+=======================================
+
+Artifact Name Composition
+-------------------------
+
+### Filenames
+
+`<--artifact_name-->@<--semantic_version-->.<--file_extension-->`
+
+Example:
+
+`electron-bootloader@1.0.1+lto.debug.jtag.bin`
+
+#### Filename Composition
+
+- Artifact Name
+  - (Platfrom Family Name (_none_) | Platform Name (`electron`))
+  - Module Name (`bootloader`)
+- Semantic Version
+  - Major Version (`1`)
+  - Minor Version (`0`)
+  - Patch Version (`1`)
+  - [_OPTIONAL_] Prerelease Tag (_none_)
+  - [_OPTIONAL_] Build Metadata 1,2,...,n (`lto`, `debug`, `jtag`)
+- File Extension (`bin`)
+
+**Rules:**
+
+- Artifacts targeting all platforms within a given platform family, shall include the platform family name and not a specific platform name (i.e. `gen3-tinker@1.1.0.bin`)
```

Comment:
Do we have such artifacts? There is no Gen 3 tinker, it's built for every platform.

Reply:
I got this from @m-mcgowan 's gist. This is forward thinking at the moment, but it sounded good.